### PR TITLE
Closes #854

### LINKED ISSUE #854: Add implementation for the Paragraph-level Simplification of Medical Texts dataset
## Adding a Dataset
- **Name:** Paragraph-level Simplification of Medical Texts
- **Description:** A paired dataset of technical medical abstracts and their plain-language summarizations.
- **Task:** SUM
- **Paper:** https://arxiv.org/abs/2104.05767
- **Data:** https://github.com/AshOlogn/Paragraph-level-Simplification-of-Medical-Texts
- **License:** CC_BY_4p0
- **Motivation:** High-quality summarization dataset for translating biomedical technical language -> layman's terms

### PR DESCRIPTION
**Pull request by Miking98**
Add the Paragraph-Level Simplification of Medical Texts dataset. Closes #854

### Checkbox

- [X] Confirm that this PR is linked to the dataset issue.
- [X] Create the dataloader script `biodatasets/my_dataset/my_dataset.py` (please use only lowercase and underscore for dataset naming).
- [X] Provide values for the `_CITATION`, `_DATASETNAME`, `_DESCRIPTION`, `_HOMEPAGE`, `_LICENSE`, `_URLs`, `_SUPPORTED_TASKS`, `_SOURCE_VERSION`, and `_BIGBIO_VERSION` variables.
- [X] Implement `_info()`, `_split_generators()` and `_generate_examples()` in dataloader script.
- [X] Make sure that the `BUILDER_CONFIGS` class attribute is a list with at least one `BigBioConfig` for the source schema and one for a bigbio schema.
- [X] Confirm dataloader script works with `datasets.load_dataset` function.
- [X] Confirm that your dataloader script passes the test suite run with `python -m tests.test_bigbio biodatasets/my_dataset/my_dataset.py`.
- [X] If my dataset is local, I have provided an output of the unit-tests in the PR (please copy paste). This is OPTIONAL for public datasets, as we can test these without access to the data files.

### CODE REVIEW

**Comment by galtay** [comment]
@Miking98 thanks for this contribution! we are in the middle of updating our contribution guidelines to support hub datasets. Can I ask that we hold off on merging this until the new guidelines are published and/or can you update your PR to include an implementation in the  `hub_repos` directory? 

this is the PR that has the new contribution guidelines https://github.com/bigscience-workshop/biomedical/pull/850

and this is an example of a PR contributing code to the `hub_repos` directory (but it wont be easily testable until the PR above is merged) https://github.com/bigscience-workshop/biomedical/pull/852

**Comment by Miking98** [comment]
Thanks for the note @galtay, makes sense! Will hold off until the new guidelines are published in that case, then will revise and submit a new pull request once updated to abide by them. Thanks!

**Comment by galtay** [comment]
hello @Miking98 thanks for your patience! we have a new `CONTRIBUTING.md` file now (https://github.com/bigscience-workshop/biomedical/blob/main/CONTRIBUTING.md) and I was wondering if you'd help us try it out. Please ping me if there are any issues and I'll help get this dataset loader in.

**Comment by Miking98** [comment]
Thanks for the note @galtay ! I just went through the revised Contributing doc and updated my pull request accordingly -- please let me know your thoughts

File: bigbio/hub/hub_repos/medparasimp/__init__.py
Comment (phlobo) [inline bigbio/hub/hub_repos/medparasimp/__init__.py:1]: Can this file be removed?

File: bigbio/hub/hub_repos/medparasimp/medparasimp.py
Comment (phlobo) [inline bigbio/hub/hub_repos/medparasimp/medparasimp.py:29]: The schema needs to be importend from `.bigbiohub` rather than bigbio (which is not available when loading from the HF hub)

File: bigbio/hub/hub_repos/medparasimp/medparasimp.py
Comment (phlobo) [inline bigbio/hub/hub_repos/medparasimp/medparasimp.py:236]: Code for source and bigbio_t2t seems to be identical, can you merge these code blocks into one to avoid duplication?

File: bigbio/hub/hub_repos/medparasimp/README.md
Comment (phlobo) [inline bigbio/hub/hub_repos/medparasimp/README.md:19]: Please update the name of the dataset

**Review by phlobo — CHANGES_REQUESTED** [review]
Thank your for the PR! Happy to merge after some very minor revisions.

**Comment by phlobo** [comment]
@mariosaenger I made some minor fixes myself, could you please review this PR?

**Review by phlobo — APPROVED** [review]

**Review by mariosaenger — APPROVED** [review]
LGTM. Thanks to all contributors.

**Pull request merged**
